spec/index.bs

```diff
@@ -101,10 +130,35 @@ partial interface WorkerNavigator {
 };
 </script>
 
-## GPUAdapter ## {#adapter}
+<script type=idl>
+[Exposed=Window]
+interface GPU {
+    // May reject with DOMException  // TODO: DOMException("OperationError")?
+    Promise<GPUAdapter> requestAdapter(optional GPURequestAdapterOptions options);
+};
+</script>
+
+## Adapters ## {#adapters}
+
+An <dfn dfn>adapter</dfn> represents an implementation of WebGPU on the system.
+Each adapter identifies both an instance of a hardware accelerator (e.g. GPU or CPU) and
+an instance of a software implementation of WebGPU on top of that accelerator.
+
+Note:
+An [=adapter=] may be a physical display adapter (GPU), but it could also be
+a software renderer.
+Different [=adapters=] could refer to different implementations on the
+same physical adapter (e.g. Vulkan and D3D12),
+or to different instances of the same physical configuration
+(e.g. if the GPU were disconnected and reconnected).
+
+### GPUAdapter ### {#gpu-adapter}
+
+A <dfn interface>GPUAdapter</dfn> refers to an [=adapter=],
+and exposes its capabilities (extensions and limits).
 
 <script type=idl>
-interface GPUAdapter : GPUObjectBase {
```

Comment:
why is this no longer a base object?

Reply:
GPUObjectBase now always has an internal `[[device]]` slot, which doesn't make sense for adapters.

It loses the label. Do you think the label is helpful?

Reply:
It's just inconsistent to not have a label here, since there is no technical ground on which we would say that the label in inapplicable to adapters.